src/utils/config.py

```diff
@@ -40,21 +61,33 @@ class ESNConfig(NamedTuple):
     input_size: int
     reservoir_size: int
     output_size: int
+    random_feature_size: int = 0
     # weight initializations
     init_weights: str = 'normal'
+    init_weights_g: str = 'normal'
+    init_weights_f: str = 'normal'
     init_weights_in: str = 'normal'
     init_weights_b: str = 'normal'
     # sparsity
     init_weights_density: float = 1.0
+    init_weights_g_density: float = 1.0
+    init_weights_f_density: float = 1.0
     init_weights_in_density: float = 1.0
     # arguments to init functions
     init_weights__args: dict = {}
+    init_weights_g__args: dict = {}
+    init_weights_f__args: dict = {}
     init_weights_b__args: dict = {}
     init_weights_in__args: dict = {}
+    # architecture parameter
+    architecture: bool = False
```

Comment:
variable naming: since these are booleans, it would be better/easier to name them something like `is_top_layer`, `is_random_feature`, etc.